packages/dataviews/src/dataforms-layouts/data-form-layout.tsx

```diff
@@ -73,12 +103,53 @@ export function DataFormLayout< Item >( {
 					return children( FieldLayout, formField );
 				}
 
+				const errorMessage =
+					( fieldDefinition?.validationSchema?.onTouched &&
+						validation.touchedFields.includes(
+							fieldDefinition.id
+						) ) ||
+					! fieldDefinition?.validationSchema?.onTouched
+						? Object.values(
+								validation.errorMessages[ formField.id ] ?? []
+						  )[ 0 ]
+						: '';
```

Comment:
What is this doing and is it the right place for this logic?

Reply:
The errorMessages dictionary contains an array of error messages for each field ID. This allows a field to have multiple error messages, such as those for password requirements.

Currently, only the first error message is passed to the field.

Reply:
I would also lean towards moving this logic to another place, could this logic be handled within the `DataFormProvider` or the `useValidation` hook?  

Also I would write this out to an `if` statement, this is rather difficult to read.